scenes/game/ai/ai_resource.gd

```diff
@@ -0,0 +1,178 @@
+## A data structure for AI manipulation. Behaves mostly like a named tuple (qv.
+## Python), in that it's like a dictionary with extra steps.
+##
+## This mainly exists so that there's specific support for object duplication
+## and value-based equality checking. By default, Godot objects cannot be
+## duplicated (and Resources only support duplication for objects whose _init
+## contains no required arguments), and always check equality by reference only.
+##
+## A class that extends AIResource should implement:
+##
+##   * _init
+##     * Takes zero required arguments.
+##     * If all arguments have their default value, the resulting object should
+##       be as close to a blank slate as possible; for example, if there are
+##       _init behaviors that perform a sync-to-original, these should *not*
+##       happen if all values are default.
+##   * copy
+##     * This is probably just a call to copy_impl but also passes in the
+##       identity of the subclass.
+
+class_name AIResource
```

Comment:
Maybe CopyableResource if you're planning to use this ina  lot of places?

Reply:
Done.